runtime/src/main/java/com/tietoevry/quarkus/resteasy/problem/javax/ConstraintViolationExceptionMapper.java

```diff
@@ -35,9 +45,41 @@ protected HttpProblem toProblem(ConstraintViolationException exception) {
     }
 
     private Violation toViolation(ConstraintViolation<?> constraintViolation) {
-        return new Violation(
-                constraintViolation.getMessage(),
-                dropMethodNameAndArgumentPositionFromPath(constraintViolation.getPropertyPath()));
+        String invalidArg = extractArgValue(constraintViolation.getPropertyPath());
+        Parameter invalidParam = getInvalidParameter(invalidArg);
+        return createViolation(constraintViolation, invalidParam);
+    }
+
+    private Parameter getInvalidParameter(String invalidArg) {
+        int paramPosition = Integer.parseInt(invalidArg.substring(3));
```

Comment:
Magic number `3`.

Reply:
You are right @pazkooda this can be named properly

Reply:
I did not comment on this topic, as it is not feasible solution in regards to custom naming provider - you cannot expect arg name to have digit/number at the end in this case. One possible approach is to look for parameters  by second segment of the path from ConstraintViolation - this will cover both cases I belive.

Reply:
That's also valid, I realized that after merging your PR into my branch and checked tests that are not passing.